test/Microsoft.ML.Tests/TensorFlowEstimatorTests.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Runtime.RunTests;
+using Microsoft.ML.Runtime.Tools;
+using Microsoft.ML.Transforms;
+using System.Collections.Generic;
+using System.IO;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Tests
+{
+    public class TensorFlowEstimatorTests : TestDataPipeBase
+    {
+        private class TestData
+        {
+            [VectorType(4)]
+            public float[] a;
+            [VectorType(4)]
+            public float[] b;
+        }
+        private class TestDataSize
+        {
+            [VectorType(2)]
+            public float[] a;
+            [VectorType(2)]
+            public float[] b;
+        }
+        private class TestDataXY
+        {
+            [VectorType(4)]
+            public float[] A;
+            [VectorType(4)]
+            public float[] B;
+        }
+        private class TestDataDifferntType
+        {
+            [VectorType(4)]
+            public string[] a;
+            [VectorType(4)]
+            public string[] b;
+        }
+
+        public TensorFlowEstimatorTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        void TestSimpleCase()
+        {
+            var modelFile = "model_matmul/frozen_saved_model.pb";
+
+            var dataView = ComponentCreation.CreateDataView(Env,
+                new List<TestData>(new TestData[] {
+                    new TestData()
+                    {
+                        a = new[] { 1.0f, 2.0f,3.0f, 4.0f },
+                        b = new[] { 1.0f, 2.0f,3.0f, 4.0f }
+                    },
+                     new TestData()
+                     {
+                         a = new[] { 2.0f, 2.0f,2.0f, 2.0f },
+                         b = new[] { 3.0f, 3.0f,3.0f, 3.0f }
+                     }
+                }));
+
+            var xyData = new List<TestDataXY> { new TestDataXY() { A = new float[4], B = new float[4] } };
+            var stringData = new List<TestDataDifferntType> { new TestDataDifferntType() { a = new string[4], b = new string[4] } };
+            var sizeData = new List<TestDataSize> { new TestDataSize() { a = new float[2], b = new float[2] } };
+            var pipe = new TensorFlowEstimator(Env, modelFile, new[] { "a", "b" }, new[] { "c" });
+
+            var invalidDataWrongNames = ComponentCreation.CreateDataView(Env, xyData);
+            var invalidDataWrongTypes = ComponentCreation.CreateDataView(Env, stringData);
+            var invalidDataWrongVectorSize = ComponentCreation.CreateDataView(Env, sizeData);
+            TestEstimatorCore(pipe, dataView, invalidInput: invalidDataWrongNames, validSchemaCantFitInput: invalidDataWrongVectorSize);
```

Comment:
>validSchemaCantFitInput [](start = 83, length = 23)

I think it's unique enough to not be in the `TestEstimatorCore`. You can just check that schema propagation works and `Fit` fails in this test.